### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by nick-pestell**
This PR forms the initial work for the `btb-phylo` repository. Which is for combining output of `btb-seq` consensus files for tree building and ultimately inclusion in ViewBovine. 

There is one main file `build_snp_matrix.py` which currently has functionality for downloading samples from s3, combining into a giant "multi_fasta" running `snp-sites` and finally `snp-dists`. I think we will also add additional functionality to build trees as an option.

The workflow involves:
1. getting all the samples to include in phylogeny (`get_samples_df()`):
- downloads the summary.csv (contains all metadata for all samples) from s3 (`summary_csv_to_df()`).
- filter the dataframe according to parameter specified at call time (**kwargs) (`filter_df()`)
- removes duplicates by choosing the submission with higher pc_mapped (`remove_duplicates()`)
2.  The multi_fasta is build in `build_multi_fasta()` and `append_multi_fasta()` which downloads the consensus sequences from the URIs detailed in samples.csv which remain after filtering and appends them to the multi_fasta_file.
3. `snp-sites` and then `snp-dists` is called.

There are two additional files. `utils.py` contains some utility functions. And `unittests.py` contains some unit tests.

### CODE REVIEW

File: build_snp_matrix.py [hunk 1]
```diff
@@ -1,5 +1,252 @@
...
+            **kwargs: 0 or more optional arguments. Names must match a 
+            column name in btb_wgs_samples.csv. If column is of type 
+            'categorical' or 'object', vales must be of type 'list' 
+            ispecifying a set of values to match against the argument 
+            name's column in btb_wgs_samples.csv. For example, 
+            'sample_name=["AFT-61-03769-21", "20-0620719"]' will include 
+            just these two samples. If column is of type 'int' or 'float',
+            values must be of type 'tuple' and of length 2, specifying a 
+            min and max value for that column. 
+
```
Comment (ellisrichardj) [inline build_snp_matrix.py:101]: I assume this is where it would be possible to generate a list for a specific clade (group column in the csv)?

Comment (ellisrichardj) [inline build_snp_matrix.py:101]: I assume this is where it would be possible to generate a list for a specific clade (group column in the csv)?

Comment (nick-pestell) [reply build_snp_matrix.py:101]: Yes, that's dealt with here, by a call to `filter_columns_categorical()`. Would just add an argument, e.g. `group=[B6-11]`

**Review by ellisrichardj — APPROVED** [review]
All looking good - nice work Nick

**Pull request merged**